pkg/automation/recommendations.go

```
@@ -43,12 +49,27 @@ func (s *googleService) ListRecommendations(project, location, recommenderID str
 		recommendations = append(recommendations, response.Recommendations...)
 		return nil
 	}
-
-	err := listCall.Pages(s.ctx, addRecommendations)
-	if err != nil {
-		return nil, err
-	}
-	return recommendations, nil
+	err := DoRequestWithRetries(func() error {
+		recommendations = nil
+		err := listCall.Pages(s.ctx, addRecommendations)
+		// Check if error is because current location is not available for getting recommendations.
+		if googleErr, ok := err.(*googleapi.Error); ok {
+			body := googleErr.Body
+			var fields map[string]interface{}
+			if parseErr := json.Unmarshal([]byte(body), &fields); parseErr != nil {
+				return err
```

Let's not return this error, it's better to return the original error.

---

it's the original error. I made a check with return here to avoid having many nested ifs.